src/main/kotlin/org/ptss/support/infrastructure/repositories/ProductRepository.kt

```diff
@@ -0,0 +1,67 @@
+package org.ptss.support.infrastructure.repositories
+
+import com.azure.data.tables.TableClient
+import com.azure.data.tables.TableServiceClientBuilder
+import jakarta.annotation.PostConstruct
+import jakarta.enterprise.context.ApplicationScoped
+import org.ptss.support.domain.models.Product
+import org.ptss.support.domain.interfaces.repositories.IProductRepository
+import org.ptss.support.infrastructure.config.AzureStorageConfig
+import org.ptss.support.infrastructure.persistence.entities.ProductEntity
+import org.slf4j.LoggerFactory
+
+@ApplicationScoped
+class ProductRepository(
+    private val azureConfig: AzureStorageConfig
+) : IProductRepository {
+    private val logger = LoggerFactory.getLogger(ProductRepository::class.java)
+    private lateinit var tableClient: TableClient
+
+    @PostConstruct
+    fun initialize() {
+        val tableServiceClient = TableServiceClientBuilder()
+            .connectionString(azureConfig.connectionString())
+            .buildClient()
+
+        // Create table if it doesn't exist
+        try {
+            tableServiceClient.createTableIfNotExists(azureConfig.tableName())
+        } catch (e: Exception) {
+            logger.error("Error creating table: ${e.message}")
+            throw e
+        }
```

Comment:
Gebruik hier executeWithLoggingAsync.

Reply:
What is executeWithLoggingAsync? Do you mean making a new method that does logging asynchronous? Should i do this everywhere?

Reply:
Check effe de code voor gebruik van `executeWithLoggingAsync`. Het is een helper function die error handling meer clean maakt. Je mag de code van `executeWithLoggingAsync` ook best veranderen verder als het de functie beter maakt.

Reply:
Found out @PostConstruct methods in Jakarta EE must be synchronous - they can't be suspend functions. The initialize() method needs to be a regular function that returns void.
hence excecute with logging async doesn't work :'(
https://docs.spring.io/spring-framework/reference/integration/scheduling.html#scheduling-annotation-support-async

Reply:
Dan zou ik de oude implementatie opnieuw gebruiken, dat maakt het ook makkelijker om te retracen waar de error vandaan kwam, ipv dat je een Azure Table Storage error krijgt, wat ook genoeg andere dingen kan zijn. 

Btw, je hebt nu ook twee keer `tableClient = tableServiceClient.getTableClient(azureConfig.tableName())` in je ProductRepository staan.